modules/common/test/helpers/volumes.go

```diff
@@ -50,3 +50,80 @@ func (tc *TestHelper) AssertVolumeMountExists(name string, subPath string, volum
 
 	return exist
 }
+
+// AssertVolumeMountPathExists - Returns true if mountPath and subPath exist in
+// the volumeMounts array.
+// Assumptions:
+//   - an empty []corev1.VolumeMount results in a failure
+//   - an empty "name" results in a failure
+//   - when no mountPath and no subPath are passed (empty string) it results in a
+//     failure
+//   - if no mountPath is provided it is skipped during the evaluation and
+//   - if no subPath is provided it is skipped during the evaluation
+//   - when both mountPath and subPath are passed, it returns true only when
+//     both are found in the volumeMount array
+func (tc *TestHelper) AssertVolumeMountPathExists(
+	name string,
+	mountPath string,
+	subPath string,
+	volumeMounts []corev1.VolumeMount,
+) bool {
+	// Early return for an empty volumeMounts list
+	if len(volumeMounts) == 0 {
+		gomega.Expect(false).To(gomega.BeTrue(), "Volume mounts list is empty")
+		return false
+	}
+
+	// Early return for an empty VolumeMount name
+	if name == "" {
+		gomega.Expect(false).To(gomega.BeTrue(), "Volume name cannot be empty")
+		return false
+	}
+
+	// Early return if both mountPath and subPath are empty strings
+	if mountPath == "" && subPath == "" {
+		gomega.Expect(false).To(gomega.BeTrue(), "Both mountPath and subPath are empty")
+		return false
+	}
+
+	// init two bool variables to represent the status of mountPath and
+	// subPath check: when the function reaches this point we have at least an
+	// input that should be checked
+	mountPathExists := false
+	subPathExists := false
+
+	if mountPath == "" {
+		// do not evaluate mountPath if not passed as input
+		// (consider it matched)
+		mountPathExists = true
+	}
+
+	if subPath == "" {
+		// do not evaluate subPath if not passed as input
+		// (consider it matched)
+		subPathExists = true
+	}
+
+	for _, v := range volumeMounts {
+		// only process the current volumeMount if the name
+		// mathes
+		if v.Name == name {
+			// mountPath is passed as input and it has not
+			// been evaluated
+			if mountPath != "" && !mountPathExists {
+				if v.MountPath == mountPath {
+					mountPathExists = true
+				}
+			}
+			// subPath is passed as input and it has not
+			// been evaluated
+			if subPath != "" && !subPathExists {
+				if v.SubPath == subPath {
+					subPathExists = true
+				}
+			}
+		}
```

Comment:
for the call of this assert, do we expect that the mountPath and the subPath are for the same entry, or do we want to cover the situation in different entries, like `mountPath: /var/lib/config-data/default` and ` subPath: keystone-api-config.json`

```
    - mountPath: /var/lib/config-data/default
      name: config-data
    - mountPath: /var/lib/kolla/config_files/config.json
      name: config-data
      readOnly: true
      subPath: keystone-api-config.json
```

if the assert is for checking `mountPath` and `subPath` for the same entry in the list, we could probably break if one of them is true?

Reply:
ok so we start the iteration if the function sees the right input (and at least we execute one loop cycle).
For each iteration, if we that both bools are set, we break the loop.  
@stuggi I'm resolving the comment but feel free to reopen if I missed something.